pkg/api/v1/atlascluster_types.go

```diff
@@ -116,6 +116,99 @@ type AtlasClusterSpec struct {
 	Project ResourceRefNamespaced `json:"projectRef"`
 
 	ClusterSpec *ClusterSpec `json:"clusterSpec,omitempty"`
```

Comment:
We can also mark ClusterSpec as optional to signify that you need one or the other.

Reply:
I believe the default is optional unless you specify required, but we can add it to be explicit